### ISSUE TITLE
Error when user code contains `class Function`

### ISSUE THREAD

**Issue by NorbertAtSAP** [istanbuljs/nyc#661]
When user code contains a class named `Function`, a name clash occurs in the instrumented code.

To reproduce, use the following file `foobar.js`:

```
"use strict";
class Function {
}
console.log('Foo bar');
```

When trying to cover this with `nyc node foobar.js`, I get the following error: 
```
ReferenceError: Function is not defined
    at C:\test\foobar.js:1:240
    at Object.<anonymous> (C:\test\foobar.js:1:662)
    at Module._compile (module.js:570:32)
    at Module.replacementCompile (C:\test\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\test\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at Object.<anonymous> (C:\test\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Function.Module.runMain (module.js:604:10)```
```

It works fine if I call my class `Function2` 

Greetings - Norbert

**Comment by bcoe** [comment]
@NorbertAtSAP thanks for the bug report, definitely an interesting one; if you have any interest in contributing, would happily accept a failing test or fix if you feel like digging into this.

We have a community slack here: http://devtoolscommunity.herokuapp.com/

**Comment by coreyfarrell** [comment]
It appears this issue was fixed in `nyc@11.5.0` and newer.  If I'm incorrect please comment and we can reopen this ticket.
